react/components/withPerformanceMeasures.tsx

```diff
@@ -0,0 +1,247 @@
+
+const SPLUNK_ENDPOINT = 'https://splunk72-heavyforwarder-public.vtex.com:8088'
+const SPLUNK_TOKEN = 'cce4a8e7-6e7a-40a0-aafb-ac45b0e271ba'
+
+const splunkLogger = new SplunkEvents()
+
+splunkLogger.config({
+  endpoint: SPLUNK_ENDPOINT,
+  token: SPLUNK_TOKEN,
+  source: 'log',
+})
+
+const { logEvent } = splunkLogger
+
+const MEASURES = [
+  {
+    name: 'from-start-to-first-render',
+    start: undefined,
+    end: 'hydration',
+  },
+  {
+    name: 'intl-polyfill',
+    start: 'intl-polyfill-start',
+    end: 'intl-polyfill-end',
+  },
+  {
+    name: 'uncritical-styles',
+    start: 'uncritical-styles-start',
+    end: 'uncritical-styles-end',
+  },
+  {
+    name: 'content-loaded',
+    start: undefined,
+    end: 'content-loaded-promise-resolved',
+  },
+  {
+    name: 'from-init-inline-js-to-first-render',
+    start: 'init-inline-js',
+    end: 'hydration',
+  },
+  {
+    name: 'from-script-start-to-first-render',
+    start: 'init-runScript',
+    end: 'hydration',
+  },
+  {
+    name: 'script-init',
+    start: 'init-runScript',
+    end: 'asyncScriptsReady-fired',
+  },
+  {
+    name: 'render-start-interval',
+    start: 'asyncScriptsReady-fired',
+    end: 'render-start',
+  },
+  {
+    name: 'first-render',
+    start: 'render-start',
+    end: 'hydration',
+  },
+]
+
+/** performance.measure throws an error if the markers don't exist.
+ * This function makes its usage more ergonomic.
+ */
+function performanceMeasure(
+  ...args: Parameters<typeof window.performance.measure>
+): PerformanceMeasure | null | undefined | void {
+  try {
+    const measure = window?.performance?.measure?.(...args)
+    if (measure as PerformanceMeasure | undefined) {
+      return measure
+    }
+    // Fix for Firefox. Performance.measure doesn't return anything it seems,
+    // but you can still get it via getEntriesByName and the like.
+    const [name] = args ?? []
+    if (typeof name !== 'string') {
+      return null
+    }
+    const entriesByName = window?.performance?.getEntriesByName?.(name)
+    const [firstEntry] = entriesByName ?? []
+    if (!isPerformanceMeasure(firstEntry)) {
+      return null
+    }
+    return firstEntry
+  } catch (e) {
+    return null
+  }
+}
+
+function isPerformanceMeasure(value: any): value is PerformanceMeasure {
+  return value?.entryType === 'measure'
+}
+
+function shouldLogPerformanceMeasures({
+  account,
+  page,
+  domain,
+}: {
+  account?: string
+  page?: string
+  domain?: string
+}) {
+  if (!account) {
+    return
+  }
+  if (domain !== 'store') {
+    return
+  }
+  const shouldDebugLogMeasures = window?.location?.search?.includes?.(
+    '__debugLogMeasures'
+  )
+  if (shouldDebugLogMeasures) {
+    return true
+  }
+
+  const DEFAULT_LOG_SAMPLING_RATE = 0.0075
+  // Allows increasing the log rate of certain accounts and pages for closer analysis.
+  const HIGHLIGHT_LOG_SAMPLING_RATE = 0.04
+  const HIGHLIGHT_ACCOUNTS = ['carrefourbr']
+  const HIGHLIGHT_PAGES = ['store.home']
```

Comment:
The change is obscured because the diff got messed up due to moving this code to a different file, but this is one of the main changes:

Increases sampling rate in general, and for the carrefourbr home in particular